core/storage/ipfs/impl/blockservice_impl.hpp

```diff
@@ -0,0 +1,38 @@
+/**
+ * Copyright Soramitsu Co., Ltd. All Rights Reserved.
+ * SPDX-License-Identifier: Apache-2.0
+ */
+
+#ifndef FILECOIN_STORAGE_IPFS_BLOCKSERVICE_IMPL_HPP
+#define FILECOIN_STORAGE_IPFS_BLOCKSERVICE_IMPL_HPP
+
+#include <memory>
+
+#include "storage/ipfs/blockservice.hpp"
+
+namespace fc::storage::ipfs {
+  class BlockServiceImpl : public BlockService {
+   public:
+    /**
+     * @brief Construct block service
+     * @todo Add exchange for remote storage
+     * @param data_store
+     */
+    explicit BlockServiceImpl(std::shared_ptr<IpfsDatastore> data_store)
+        : local_storage_{std::move(data_store)} {}
```

Comment:
move it to cpp and add check BOOST_ASSERT_MSG(local_storage_ != nullptr, "message")